Source/XUnitExtensions/Lit/OutputCheckCommand.cs

```diff
@@ -185,6 +185,19 @@ private record CheckNotLiteral(string File, int LineNumber, string Literal) : Ch
     return result!;
   }
 
+  public static IList<CheckDirective> ParseCheckFile(string fileName) {
+    var result = File.ReadAllLines(fileName)
+      .Select((line, index) => CheckDirective.Parse(fileName, index + 1, line))
+      .Where(e => e != null).Cast<CheckDirective>()
+      .Select(e => e!)
```

Comment:
I think you're right - I added this because I swear the compiler forced me to at some point, but in the current build configuration there's no warning or error without this line. It's possible that's because of this annotation on the return type parameter of `Cast` that might be new:

```
#nullable disable // there's no way to annotate the connection of the nullability of TResult to that of the source
                TResult
#nullable restore
```

It should be impossible for `e!` to throw by construction so I'm comfortable removing it, especially in testing-only code.